src/update.ts

```diff
@@ -0,0 +1,163 @@
+import { Update as U, Message as M } from './core/types/typegram'
+
+/**
+ * Update types to use with `Context<Update>`.
+ * The exact structure of this export may change in the future, or be removed entirely.
+ * @experimental
+ */
+// eslint-disable-next-line @typescript-eslint/no-namespace
+export namespace Update {
+  export interface Message extends U.MessageUpdate {}
+  export interface EditedMessage extends U.EditedMessageUpdate {}
+  export interface ChannelPost extends U.ChannelPostUpdate {}
+  export interface EditedChannelPost extends U.EditedChannelPostUpdate {}
+  export interface InlineQuery extends U.InlineQueryUpdate {}
+  export interface ChosenInlineResult extends U.ChosenInlineResultUpdate {}
+  export interface CallbackQuery extends U.CallbackQueryUpdate {}
+  export interface ShippingQuery extends U.ShippingQueryUpdate {}
+  export interface PreCheckoutQuery extends U.PreCheckoutQueryUpdate {}
+  export interface Poll extends U.PollUpdate {}
+  export interface PollAnswer extends U.PollAnswerUpdate {}
+  export interface MyChatMember extends U.MyChatMemberUpdate {}
+  export interface ChatMember extends U.ChatMemberUpdate {}
+  export interface ChatJoinRequest extends U.ChatJoinRequestUpdate {}
+
+  export type All =
+    | Message
+    | EditedMessage
+    | ChannelPost
+    | EditedChannelPost
+    | InlineQuery
+    | ChosenInlineResult
+    | CallbackQuery
+    | ShippingQuery
+    | PreCheckoutQuery
+    | Poll
+    | PollAnswer
+    | MyChatMember
+    | ChatMember
+    | ChatJoinRequest
+
+  // eslint-disable-next-line @typescript-eslint/no-namespace
+  export namespace Message {
```

Comment:
There's no equivalent for edited messages and channel posts. Wouldn't generics help? `Update.Message<Message.Text>`

Reply:
This would be consistent with `Context<Update.Message>`.

Reply:
Fair, need more time on this.

Reply:
Done, made `Message`, `EditedMessage`, `ChannelPost`, and `EditedChannelPost` generic. I've just used `Message` from typegram now. I wanted to reassign, but it increases the file size by a lot for not much benefit.